src/node_data.js

```diff
@@ -117,17 +117,18 @@ const getData = function(node) {
 /**
  * Imports node and its subtree, initializing caches.
  *
- * @param {?Node} node The Node to import.
+ * @param {!Node} node The Node to import.
+ * @param {?Node} parentNode the node's parent.
  * @return {!NodeData} The NodeData for this Node.
  */
-const importNode = function(node) {
+const importInternal = function(node, parentNode) {
```

Comment:
Could do

`const parentNode = opt_parentNode || node.parentNode`

and not have the `importInternal` function. Or maybe:

Could also do `const parentData = opt_parentData || getData(node.parentNode)`